engine/include/cubos/engine/rendering/deferred/deferred_renderer.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef CUBOS_RENDERING_DEFERRED_DEFERRED_RENDERER_HPP
+#define CUBOS_RENDERING_DEFERRED_DEFERRED_RENDERER_HPP
+
+#include <vector>
+
+#include "cubos/core/gl/vertex.hpp"
+#include <cubos/core/gl/render_device.hpp>
+#include "cubos/engine/rendering/renderer.hpp"
+
+#define CUBOS_DEFERRED_RENDERER_MAX_SPOT_LIGHT_COUNT 128
+#define CUBOS_DEFERRED_RENDERER_MAX_DIRECTIONAL_LIGHT_COUNT 128
+#define CUBOS_DEFERRED_RENDERER_MAX_POINT_LIGHT_COUNT 128
+
+namespace cubos::engine::rendering
```

Comment:
Make this namespace `cubos::engine::gl::deferred` and rename the class from `DeferredRenderer` to `Renderer` in order to be consistent with other namespace naming.